bin/_tag.sh

```diff
@@ -24,37 +24,37 @@ head_root_tag() {
     if clean_head ; then
         clean_head_root_tag
     else
-        name="$(echo $USER | sed "s/[^[:alnum:].-]//g")"
+        name=$(echo $USER | sed 's/[^[:alnum:].-]//g')
         echo "dev-$(git_sha_head)-$name"
     fi
 }
 
 clean_head_root_tag() {
     if clean_head ; then
-        if [ "$(named_tag)" != "undefined" ]; then
+        if [ "$(named_tag)" != undefined ]; then
             echo "$(named_tag)"
         else
             echo "git-$(git_sha_head)"
         fi
     else
-        echo "Commit unstaged changes." >&2
+        echo 'Commit unstaged changes.' >&2
         exit 3
     fi
 }
 
 validate_tag() {
-    file="$1"
+    file=$1
     shift
 
-    image="$1"
+    image=$1
     shift
 
-    sha="$1"
+    sha=$1
     shift
 
-    dockerfile_tag=$(grep -oe $image':[^ ]*' $file) || true
+    dockerfile_tag=$(grep -oe $image':[^ ]*' "$file") || true
```

Comment:
should $image be double quoted here?

Reply:
The only place where I see validate_tag being called is on line 72 in _tag.sh, where the value is gcr.io/linkerd-io/go-deps. Besides, I believe whitespaces are not allowed in that context, is it? That's why I figured $image would not be expected to contain whitespaces (=no quotes necessary).

Reply:
I think you're right, but it seems like a good practice/standard to always quote variables in this type of scenario to be safe.  Is there any reason not to?

To be perfectly honest, I don't know very much about shell scripting and I lean *very* heavily on https://www.shellcheck.net/ for best practices.  So even if it doesn't make a difference in this case, I feel a lot more comfortable if I can make shellcheck happy.

Reply:
I guess it is sort of a philosophical question; one side prefers fail-early principles and do not quote where content that should not be quote is, while the other side says pass on whatever in a safe manner and leave it for the end consumer to decide what is ok an what is not. To me, as long as there is a clear understanding of why and why not something was quoted, I see benefits with any of those approaches.
I will update the patchset with your suggestion!